pkg/skr/skr.go

```diff
@@ -88,30 +90,39 @@ func SecureKeyRelease(identity common.Identity, certState attest.CertState, SKRK
 		return nil, errors.Wrapf(err, "attestation failed")
 	}
 
-	// 2. Interact with Azure Key Vault. The REST API of AKV requires
-	//     authentication using an Azure authentication token.
+	var ResourceIDTemplate string
+
+	// If endpoint contains managedhsm, request a token for managedhsm
+	// resource; otherwise for a vault
+	if ResourceIDTemplate = ResourceIdVault; strings.Contains(SKRKeyBlob.AKV.Endpoint, "managedhsm") {
+		ResourceIDTemplate = ResourceIdManagedHSM
+		logrus.Infof("Requesting token from %s", ResourceIDTemplate)
+	}
 
 	// retrieve an Azure authentication token for authenticating with AKV
 	if SKRKeyBlob.AKV.BearerToken == "" {
-		logrus.Info("Retrieving Azure authentication token...")
-		var ResourceIDTemplate string
-		// If endpoint contains managedhsm, request a token for managedhsm
-		// resource; otherwise for a vault
-		if strings.Contains(SKRKeyBlob.AKV.Endpoint, "managedhsm") {
-			logrus.Info("Requesting token for managedhsm...")
-			ResourceIDTemplate = ResourceIdManagedHSM
+		ctx := context.TODO()

```

Comment:
Same here. Would it be worth adding a timeout to `ctx`?

Reply:
done